### PR TITLE
feat: Add layout support for inputs

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
This PR adds support for grouping inputs based on a "layout" property.

### CODE REVIEW

File: packages/ui/src/views/run-pipeline/utils/__tests__/inputs-utils.test.tsx
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/run-pipeline/utils/__tests__/inputs-utils.test.tsx:1]: can we also add a test for empty layout, when nothing is passed?

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts [hunk 1]
```diff
@@ -4,19 +4,95 @@ import * as z from 'zod'
...
+}: {
+  pipelineInputs: Record<string, any>
   options: { prefix?: string }
-): IInputDefinition[] {
-  const formInputs: IInputDefinition[] = []
+  pipelineInputLayout?: InputLayout
+}): IInputDefinition[] {
+  /**
+   * Pre-process inputs for valid layout.
+   * Rule 1 - If there are duplicate keys in the layout, we skip the layout and return all inputs flat
```
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:21]: we should not skip the entire layout, we should only skip the repeated inputs (show them only once)

File: packages/ui/src/views/run-pipeline/utils/__tests__/inputs-utils.test.tsx
Comment (vardanbansal-harness) [reply packages/ui/src/views/run-pipeline/utils/__tests__/inputs-utils.test.tsx:1]: Done.

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts [hunk 2]
```diff
@@ -4,19 +4,95 @@ import * as z from 'zod'
...
+   * Rule 2 - ...
+   */
+  const duplicateKeys = validateUniqueInputKeysInLayout(pipelineInputLayout)
+
+  // If duplicates found, skip layout and return all inputs flat
+  if (validateUniqueInputKeysInLayout(pipelineInputLayout).length > 0) {
```
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:27]:
```suggestion
  if (duplicateKeys.length > 0) {
```

Comment (vardanbansal-harness) [reply packages/ui/src/views/run-pipeline/utils/inputs-utils.ts]: Done.

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts [hunk 3]
```diff
@@ -4,19 +4,95 @@ import * as z from 'zod'
...
+
+export const validateUniqueInputKeysInLayout = (layout: InputLayout): string[] => {
+  const inputOccurrences = new Map<string, number>()
+
+  const traverse = (layout: InputLayout) => {
+    layout.forEach(item => {
+      if (typeof item === 'string') {
+        inputOccurrences.set(item, (inputOccurrences.get(item) || 0) + 1)
+      } else {
+        traverse(item.items)
```
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:84]:
```suggestion
        item.items && item.items.length > 0 && traverse(item.items)
```

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts [hunk 4]
```diff
@@ -4,19 +4,95 @@ import * as z from 'zod'
...
   })
 
-  return formInputs
+  return [...inputsFromLayout, ...remainingInputs]
+}
+
+export const validateUniqueInputKeysInLayout = (layout: InputLayout): string[] => {
+  const inputOccurrences = new Map<string, number>()
+
+  const traverse = (layout: InputLayout) => {
```
Comment (abhinavrastogi-harness) [inline packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:79]: can this be a pure function outside of `validateUniqueInputKeysInLayout` as well?

Comment (jake-bassett) [reply packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:79]: I agree this feels cleaner and allows reuse, but I'm curious what your reason for this suggestion is. Same as I listed or something else?

Comment (vardanbansal-harness) [reply packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:79]: Also done.

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts [hunk 5]
```diff
@@ -4,19 +4,95 @@ import * as z from 'zod'
...
+  pipelineInputs: Record<string, any>
   options: { prefix?: string }
-): IInputDefinition[] {
-  const formInputs: IInputDefinition[] = []
+  pipelineInputLayout?: InputLayout
+}): IInputDefinition[] {
+  /**
+   * Pre-process inputs for valid layout.
+   * Rule 1 - If there are duplicate keys in the layout, we skip the layout and return all inputs flat
+   * Rule 2 - ...
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply packages/ui/src/views/run-pipeline/utils/inputs-utils.ts]: Done, added to the list of rules.

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply packages/ui/src/views/run-pipeline/utils/inputs-utils.ts]: Done, added it as a rule

```
* Rule 1 - If input keys are duplicated in the layout, only the first occurrence is considered
```

File: packages/ui/src/views/run-pipeline/utils/inputs-utils.ts [hunk 6]
```diff
@@ -73,26 +45,62 @@ export function pipelineInputs2FormInputs({
...
 
-  const traverse = (layout: InputLayout) => {
-    layout.forEach(item => {
-      if (typeof item === 'string') {
-        inputOccurrences.set(item, (inputOccurrences.get(item) || 0) + 1)
-      } else {
-        traverse(item.items)
+    // If group has no title, flatten its items
+    if (!item.title && item.items && item.items.length > 0) {
+      return processLayout(item.items, pipelineInputs, options, processedInputKeys)
```
Comment (jake-bassett) [inline packages/ui/src/views/run-pipeline/utils/inputs-utils.ts:72]: Just a nit and maybe more of a preference, so not a big deal either way, but you could assign this line to a `const` before the `if` and return it here if we hit this branch else assign to `inputs` below if not.
